tool-wrappers/ocamllsp

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+set -eu
+
+# Script to choose an appropriate instance of ocamllsp. If the project has a
+# lockdir then ocamllsp will be installed and run as a dev-tool. Otherwise the
+# script will try to launch an opam installation of ocamllsp.
+
+if [ -d dune.lock ]; then
+
+    OCAMLFORMAT_TARGET=_build/_private/default/.dev-tool/ocamlformat/ocamlformat/target
+
+    if [ ! -f $OCAMLFORMAT_TARGET/cookie ]; then
+        # Make sure that the ocamlformat dev tool is installed as it's needed by
+        # ocamllsp. There's currently no command that just installs ocamlformat so
+        # we need to run it and ignore the result.
+        dune tools exec ocamlformat -- --help > /dev/null
+    fi
+
+    # Add ocamlformat to the environment in which ocamllsp runs so ocamllsp can invoke ocamlformat.
+    export PATH=$PWD/$OCAMLFORMAT_TARGET/bin:$PATH
+
+    # Build and run ocamllsp.
+    dune tools exec ocamllsp -- $@
```

Comment:
On one side of the `if/else` the `PATH` is exported, on the other side it is passed as environment to `ocamllsp`. It would be nicer if it were consistent.